drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -4181,7 +4181,9 @@ void RasterizerSceneGLES3::render_scene(const Transform &p_cam_transform, const
 	state.ubo_data.shadow_dual_paraboloid_render_zfar = 0;
 	state.ubo_data.opaque_prepass_threshold = 0.99;
 
-	p_cam_projection.get_viewport_size(state.ubo_data.viewport_size[0], state.ubo_data.viewport_size[1]);
+	Vector2 vp_half_exts = p_cam_projection.get_viewport_half_extents();
+	state.ubo_data.viewport_size[0] = vp_half_exts.x;
```

Comment:
@clayjohn , @reduz 
Should this be double the half extents? (I have duplicated the old behaviour, but I assume the old could be a bug)

Reply:
Looking at #19900 I now believe this could be a 'double bug'. i.e. it is now incorrectly set to half the viewport extents, rather than the full extents, _however_ the docs say this should be the width and height of the viewport in pixels.

If so then the original may have been a mistake thinking that the old CameraMatrix::get_viewport_size returned pixels.

I can't find anything in the standard shaders glsl referring to viewport_size (aside from screen space reflections which seems to replace it with something different) so there might be an argument for changing this to the actual viewport width and height in pixels. I'll try it out and see if anything looks amiss.